stubs/nvml_stubs.c

```diff
@@ -273,7 +273,7 @@ stub_nvml_device_get_handle_by_index(value ml_interface, value ml_index)
 
     unsigned int deviceSize = sizeof(nvmlDevice_t);
     ml_device = caml_alloc_string(deviceSize);
-    memcpy(String_val(ml_device), &device, deviceSize);
+    memcpy((char *) String_val(ml_device), &device, deviceSize);
```

Comment:
please don't override it like this. String is immutable. If we want to change it then use `Bytes_val`. String_val is readonly for a reason.

Reply:
Is `caml_copy_string` not possible because `device` is not a null-terminated string? Yes, what Edwin said looks good:
>caml_alloc_initialized_string(n, p) returns a byte sequence (or string) value of length n bytes. The value is initialized from the n bytes starting at address p.

Reply:
This is about the `const` warnings:

```
runtime/caml/mlvalues.h:#define String_val(x) ((const char *) Bp_val(x))
runtime/caml/mlvalues.h:#define Bytes_val(x) ((unsigned char *) Bp_val(x))
```

The `const` in `String_val` is new and causes the warning; will switch to `Bytes_val`.

Reply:
Please use `caml_alloc_initialized_string` instead as suggested in my updated comment, it does what the binding tries to do here manually without having to resort to various assumptions on strings vs bytes.